app/models/partner.rb

```diff
@@ -203,6 +215,21 @@ def contact_person
     }
   end
 
+  def agency_info
+    return @agency_info if @agency_info
+
+    return {} if profile.blank?
+
+    @agency_info = {
+      address: [profile.address1, profile.address2].reject(&:blank?).join(' '),
+      city: profile.city,
```

Comment:
Another minor functionality quibble -- can we join those two addresses with a comma instead of a space?